Code/Source/RobotImporter/URDF/CollidersMaker.cpp

```diff
@@ -0,0 +1,346 @@
+                    &LmbrCentral::BoxShapeComponentRequests::SetBoxDimensions,
+                    URDF::TypeConversions::ConvertVector3(boxGeometry->dim));
+                entity->Deactivate();
+            }
+            break;
+        case urdf::Geometry::CYLINDER:
+            {
+                entity->CreateComponent(LmbrCentral::EditorCylinderShapeComponentTypeId);
+                auto cylinderGeometry = std::dynamic_pointer_cast<urdf::Cylinder>(geometry);
+                entity->Activate();
+                LmbrCentral::CylinderShapeComponentRequestsBus::Event(
+                    entityId, &LmbrCentral::CylinderShapeComponentRequests::SetHeight, cylinderGeometry->length);
+                LmbrCentral::CylinderShapeComponentRequestsBus::Event(
+                    entityId, &LmbrCentral::CylinderShapeComponentRequests::SetRadius, cylinderGeometry->radius);
+                entity->Deactivate();
+            }
+            break;
+        default:
+            AZ_Warning("AddCollider", false, "Unsupported collider geometry type, %d", geometry->type);
+            break;
+        }
+    }
+    AZ::IO::Path CollidersMaker::GetFullURDFMeshPath(AZ::IO::Path modelPath, AZ::IO::Path meshPath)
+    {
+        modelPath.RemoveFilename();
+        AZ::StringFunc::Replace(meshPath.Native(), "package://", "", true, true);
+        modelPath /= meshPath;
+
+        return modelPath;
+    }
+
+    void CollidersMaker::ProcessMeshes(BuildReadyCallback notifyBuildReadyCb)
+    {
+        m_buildThread = AZStd::thread(
+            [this, notifyBuildReadyCb]()
+            {
+                AZ_Printf("CollisionMaker", "Waiting for URDF assets...");
+                while (!m_meshesToBuild.empty() && !m_stopBuildFlag)
+                {
+                    {
+                        AZStd::lock_guard lock{ m_buildMutex };
+                        for (auto iter = m_meshesToBuild.begin(); iter != m_meshesToBuild.end(); iter++)
+                        {
+                            AZ::Data::AssetId assetId;
+                            AZ::Data::AssetType assetType = AZ::AzTypeInfo<PhysX::Pipeline::MeshAsset>::Uuid();
+                            AZ::Data::AssetCatalogRequestBus::BroadcastResult(
+                                assetId, &AZ::Data::AssetCatalogRequests::GetAssetIdByPath, iter->Native().c_str(), assetType, false);
+                            if (assetId.IsValid())
+                            {
+                                AZ_Printf("CollisionMaker", "Asset %s found and valid...", iter->Native().c_str());
+                                m_meshesToBuild.erase(iter--);
+                            }
+                        }
+                    }
+                    if (!m_meshesToBuild.empty())
+                    {
+                        AZStd::this_thread::sleep_for(AZStd::chrono::milliseconds(50));
+                    }
+                }
+
+                AZ_Printf("CollisionMaker", "All URDF assets ready!");
```

Comment:
`AZ_Printf` doesn't automatically add newlines, so you probably want to add newlines to end of each AZ_Printf.

Reply:
Thanks for that hint!